rewrite-json/src/main/java/org/openrewrite/json/AddKeyValue.java

```diff
@@ -0,0 +1,132 @@
+        JsonPathMatcher pathMatcher = new JsonPathMatcher(keyPath);
+
+        return new JsonIsoVisitor<ExecutionContext>() {
+
+            @Override
+            public Json.JsonObject visitObject(Json.JsonObject obj, ExecutionContext context) {
+                obj = super.visitObject(obj, context);
+
+                if (pathMatcher.matches(getCursor()) && objectDoesNotContainKey(obj, key)) {
+
+                    boolean jsonIsEmpty = obj.getMembers().isEmpty() || obj.getMembers().get(0) instanceof Json.Empty;
+                    Space space = jsonIsEmpty ? Space.EMPTY : obj.getMembers().get(0).getPrefix();
+
+                    JsonRightPadded<JsonKey> newKey = rightPaddedKey();
+                    Json.Literal newValue = valueLiteral();
+                    Json newMember = new Json.Member(randomId(), space, Markers.EMPTY, newKey, newValue);
+
+                    List<Json> members = jsonIsEmpty ? new ArrayList<>() : obj.getMembers();
+
+                    if (prepend) {
+                        members.add(0, newMember);
+                    } else {
+                        members.add(newMember);
+                    }
+
+                    return obj.withMembers(members);
+                }
+                return obj;
+            }
+
+            @NotNull
+            private Json.Literal valueLiteral() {
+                return new Json.Literal(randomId(), Space.build(" ", emptyList()), Markers.EMPTY, value, unQuote(value));
+            }
+
+            @NotNull
+            private JsonRightPadded<JsonKey> rightPaddedKey() {
+                return new JsonRightPadded<>(
+                    new Json.Literal(randomId(), Space.EMPTY, Markers.EMPTY, "\"" + key + "\"", key),
+                    Space.EMPTY, Markers.EMPTY
+                );
+            }
+
+            private String unQuote(String value) {
+                if (value.startsWith("'") || value.startsWith("\"")) {
+                    value = value.substring(1, value.length() - 1);
+                }
+                return value;
+            }
+
+            private boolean objectDoesNotContainKey(Json.JsonObject obj, String key) {
+                for (Json member : obj.getMembers()) {
+                    if (member instanceof Json.Member) {
+                        if (keyMatches(((Json.Member) member).getKey(), key)) {
+                            return false;
+                        }
+                    }
+                }
+                return true;
+            }
+
+            private boolean keyMatches(JsonKey jsonKey, String key) {
+                if (jsonKey instanceof Json.Literal) {
+                    return key.equals(((Json.Literal) jsonKey).getValue());
+                } else if (jsonKey instanceof Json.Identifier) {
+                    return key.equals(((Json.Identifier) jsonKey).getName());
+                }
+                throw new IllegalStateException("Key is not 'Json.Literal' or 'Json.Identifier': " + jsonKey);
```

Comment:
hey guys so I'm not sure about this peace of code, I've only ever encountered `Json.Literal` as a key in testing, but `Json.Identifier` also implements `JsonKey` so that's why it's here. lmc if this is redundant

Reply:
Whenever we're uncertain of a change, we fallback to making no change at all. That way we ensure we [do no harm](https://docs.openrewrite.org/authoring-recipes/recipe-conventions-and-best-practices).